authorization/src/lib.rs

```diff
@@ -5,39 +5,37 @@ extern crate std;
 
 mod test;
 
-use soroban_sdk::{contractimpl, contracttype, BigInt, Env, IntoVal, Symbol};
+use soroban_sdk::{contractimpl, contracttype, symbol, BigInt, Env, IntoVal};
 use soroban_sdk_auth::{
     check_auth, NonceAuth, {Identifier, Signature},
 };
 
-#[derive(Clone)]
 #[contracttype]
 pub enum DataKey {
-    Acc(Identifier),
+    SavedNum(Identifier),
     Nonce(Identifier),
     Admin,
 }
 
 fn read_nonce(e: &Env, id: Identifier) -> BigInt {
     let key = DataKey::Nonce(id);
-    if let Some(nonce) = e.contract_data().get(key) {
-        nonce.unwrap()
-    } else {
-        BigInt::zero(e)
-    }
+    e.contract_data()
+        .get(key)
+        .unwrap_or_else(|| Ok(BigInt::zero(e)))
+        .unwrap()
 }
-struct WrappedAuth(Signature);
 
-impl NonceAuth for WrappedAuth {
+struct NonceForSignature(Signature);
+
+impl NonceAuth for NonceForSignature {
     fn read_nonce(e: &Env, id: Identifier) -> BigInt {
         read_nonce(e, id)
     }
 
     fn read_and_increment_nonce(&self, e: &Env, id: Identifier) -> BigInt {
         let key = DataKey::Nonce(id.clone());
         let nonce = Self::read_nonce(e, id);
-        e.contract_data()
-            .set(key, nonce.clone() + BigInt::from_u32(e, 1));
+        e.contract_data().set(key, &nonce + 1);
```

Comment:
BigInt's are reasonably interoperable with most number types, so we can use `+ 1` here, and we can also use a ref instead of a clone.